fs-storage/src/file_storage.rs

```diff
@@ -234,6 +271,23 @@ where
     }
 }
 
+impl<K, V> Monoid<V> for FileStorage<K, V>
```

Comment:
As you can see, this Monoid implementation does not use any property or value of the FileStorage. It doesn't even use the key. It only relies on the properties of the `V` value. In such a case, it's probably better to remove the Monoid implementation for FileStorage.

Instead, add a trait condition for V in `merge_from` like

```rust
    fn merge_from(&mut self, other: impl AsRef<BTreeMap<K, V>>) -> Result<()>
        where V: Monoid 
        {
```

And the the `resolved_value` can be decided based on the Monoid implementation of `V`. In fact, this approach is closer to your idea of using Monoid as a helper function for `FileStorage`.

Reply:
Done;
Currently, I only implemented monoid for integers and strings. Let me know if we need more.